src/incubator/Dialog/__tests__/index.new.spec.tsx

```diff
@@ -98,4 +99,11 @@ describe('Incubator.Dialog sanity checks', () => {
     expect(await dialogDriver.getModal().isVisible()).toBeFalsy();
     expect(onDismiss).toHaveBeenCalledTimes(1);
   });
+
+  it('Modal exists even when not visible (WOAUILIB-3606)', async () => {
+    const onDismiss = jest.fn();
+    const component = <TestCase2 onDismiss={onDismiss}/>;
+    const {dialogDriver} = getDriver(component);
+    expect(dialogDriver.getModal().exists()).toBeTruthy();
```

Comment:
Maybe we should check that is it not visible first? If some changes the TestCase to be visible for some reason this still passes.
```suggestion
    expect(dialogDriver.getModal().exists()).toBeTruthy();
    expect(dialogDriver.getModal().isVisible()).toBeFalsy())
```